src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2524,7 +2540,7 @@ private bool TryGetReturnType(out TypeWithAnnotations type, out FlowAnalysisAnno
                 return false;
             }
 
-            var delegateOrMethod = _delegateInvokeMethod ?? method;
+            var delegateOrMethod = _useDelegateInvokeReturnType ? _delegateInvokeMethod! : method;
```

Comment:
nit: we should probably also have an assertion (not just a suppression) since this is enforced elsewhere

Reply:
We could although we're dereferencing the result immediately.

Reply:
Ah, had missed that. Sounds okay then